compiler-rt/lib/rtsan/rtsan_assertions.h

```diff
@@ -12,10 +12,19 @@
 
 #pragma once
 
+#include "rtsan/rtsan.h"
 #include "rtsan/rtsan_context.h"
-#include "rtsan/rtsan_diagnostics.h"
 
 namespace __rtsan {
 
-void ExpectNotRealtime(Context &context, const DiagnosticsInfo &info);
+template <typename OnViolationAction>
+void ExpectNotRealtime(Context &context, OnViolationAction &&on_violation) {
```

Comment:
Don't 100% know what is right here, but I have been using FunctionCase for lambdas and other callables. Leaving it to you to determine if you want to change it or not 

(and we should get clarity on what is appropriate, our tests use a lot of `auto Func = [](){}`)